go/proxyd/proxyd.go

```diff
@@ -194,12 +206,17 @@ func Start(config *Config) error {
 		rpcCache,
 	)
 
-	if config.Metrics != nil && config.Metrics.Enabled {
+	if config.Metrics.Enabled {
 		addr := fmt.Sprintf("%s:%d", config.Metrics.Host, config.Metrics.Port)
 		log.Info("starting metrics server", "addr", addr)
 		go http.ListenAndServe(addr, promhttp.Handler())
 	}
 
+	// To allow integration tests to cleanly come up, wait
+	// 10ms to give the below goroutines enough time to
```

Comment:
This seems prone to race conditions depending on the size of the box that the tests run on but should be ok